ocs_ci/ocs/deployment_openshift_logging.py

```diff
@@ -0,0 +1,362 @@
+
+    Example:
+        cl_create_catalogsourceconfig(
+        constants.CL_CSC_YAML, 'cluster-logging-operator')
+
+    """
+
+    create_csc = ocp.OCP(
+        kind=constants.CATALOG_SOURCE_CONFIG, namespace='openshift-marketplace'
+    )
+    clusterlogging_csc = ocp.OCP(
+        kind=constants.CATALOG_SOURCE_CONFIG, namespace='openshift-logging'
+    )
+
+    create_csc.create(yaml_file=yaml_file)
+    try:
+        clusterlogging_csc.get(resource_name, out_yaml_format=True)
+        logger.info(f"The CatalogSourceConfig is created Successfully")
+    except CommandFailed:
+        logger.error(f'The resource is not found')
+        return False
+    return True
+
+
+def create_clusterlogging_subscription(yaml_file):
+    """
+    Creation of subscription for clusterlogging to subscribe
+    a namespace to an operator
+
+    Args:
+        yaml_file : Contains yaml file to create subscription for
+                    the namespace
+        resource_name (str) : Name of the subscription
+
+    Returns:
+        subscription (dict): Contains all the details of the subscription.
+
+    Example:
+        cl_create_subscription(yaml_file=constants.CL_SUB_YAML)
+
+    """
+
+    clusterlogging_subscription = ocp.OCP(
+        kind=constants.SUBSCRIPTION, namespace='openshift-logging'
+    )
+    subscription = clusterlogging_subscription.create(
+        yaml_file, out_yaml_format=True
+    )
+    logger.info(subscription)
+    subscription_info = clusterlogging_subscription.get(
+        resource_name='cluster-logging', out_yaml_format=True
+    )
+    if subscription_info:
+        logger.info(f"The Subscription is created successfully")
+    else:
+        logger.error(f"The subscription is not installed properly")
+    return subscription_info
+
+
+def create_instance_in_clusterlogging(sc_name):
+    """
+    Creation of instance for clusterlogging that creates PVCs,
+    ElasticSearch, curator fluentd and kibana pods
+
+    Args:
+        sc_name (str) : Storage class name to create PVCs
+
+    Returns:
+        logging_instance (dict): Contains all detailed information of the
+        instance such as pods that got created, its resources and limits
+        values, storage class and size details etc.
+
+    """
+    inst_data = templating.load_yaml_to_dict(constants.CL_INSTANCE_YAML)
+    inst_data['spec']['logStore']['elasticsearch']['storage']['storageClassName'] = sc_name
+    inst_data['spec']['logStore']['elasticsearch']['storage']['size'] = "15Gi"
+    helpers.create_resource(wait=False, **inst_data)
+    oc = ocp.OCP('v1', 'ClusterLogging', 'openshift-logging')
+    logging_instance = oc.get(resource_name='instance', out_yaml_format='True')
+    if logging_instance:
+        logger.info("Successfully created instance for cluster-logging")
+        logger.info(logging_instance)
+    else:
+        logger.error("Instance for clusterlogging is not created properly")
+    return logging_instance
+
+
+@retry(IndexError, 5, 20, 2)
```

Comment:
@Shrivaibavi why are we retrying for 5 times. Any specific reason for this ?

Reply:
Removed that.. Pls check the new commit